beacon-chain/node/node.go

```diff
@@ -61,6 +61,11 @@ const testSkipPowFlag = "test-skip-pow"
 // 128MB max message size when enabling debug endpoints.
 const debugGrpcMaxMsgSize = 1 << 27
 
+type serviceFlagOpts struct {
```

Comment:
I think this pattern came from a thread in discord about a code UX issue raised by Nishant, but I forget the exact problem. Could you add a comment about why this type exists?